.travis.yml

```diff
@@ -1,16 +1,54 @@
 language: python
+
 python:
   - 3.6
 
+os:
+  - linux
+  - linux-ppc64le
+
+env:
+  - DOCKERFILE=Dockerfile IMAGE=max-base
+  - DOCKERFILE=Dockerfile.pai IMAGE=max-base-powerai
+
+matrix:
+  exclude:
+  - env: DOCKERFILE=Dockerfile IMAGE=max-base
+    os: linux-ppc64le
+
+dist: xenial
+
 services:
   - docker
 
+before_install:
+  - if [[ "$TRAVIS_OS_NAME" == "linux-ppc64le" ]]; then
+       ARCH=ppc64le;
+    fi
+  - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then
+       ARCH=x86_64;
+    fi
+  - mkdir -p ~/.docker
+  - echo '{"experimental":"enabled"}' | tee ~/.docker/config.json 
+  - sudo service docker restart
+
 install:
-  - docker build -t max-base .
+  - docker build -f "$DOCKERFILE" -t "$REPO"/"$IMAGE":"$ARCH"-latest .
 
 before_script:
-  - docker run -it --rm -d -p 5000:5000 max-base
+  - docker run -it --rm -d -p 5000:5000 "$REPO"/"$IMAGE":"$ARCH"-latest
   - sleep 10
 
 script:
-  - docker ps | grep max-base
+  - docker ps | grep "$IMAGE"
+
+after_success:
+  - if [[ "$IMAGE" != "max-base" && "$TRAVIS_PULL_REQUEST" == "false" ]]; then
+      docker login -u "$REPO_USER" -p "$REPO_PASS";
+      docker push "$REPO"/"$IMAGE":"$ARCH"-latest;
+      docker pull "$REPO"/"$IMAGE":ppc64le-latest;
+      docker pull "$REPO"/"$IMAGE":x86_64-latest;
```

Comment:
IIUC this will not work on the first run of this file since only one of the above two pulls will succeed.

Reply:
This is correct. The build will succeed regardless and the multiarch image will be pushed when the other arch build finishes. Also, if we manually push initial images into the repo, the first automated build will grab that. 

Now that I think about it, I believe the multiarch manifest references the component images by name/tag and not by hash. If that's the case, we might be able to manually push the multiarch manifest once, and as long as the component images keep the same name, the multiarch image will stay up to date. I can verify this.

(Note that if there's any need to tag multiarch images with a specific build or commit number instead of just `latest` we'll still need to push manifests in each build)

Reply:
Unfortunately I confirmed this isn't true. The manifest locks to specific shas for the component images, so a new manifest needs to be pushed with each build.

A cleaner way to do this might be to add a separate manifest creation job as a new stage that depends on the previous jobs, however, and make the `docker push` for each image build required for job success.